### PR TITLE
[release-1.6] Fix config for using custom CAs.

### PR DESCRIPTION
**Pull request by myidpt**
Cherry-pick of https://github.com/istio/istio/pull/24625.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ X ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ X ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

**Comment by jacob-delgado** [comment]
Waiting on resolution of https://github.com/istio/istio/pull/24625

**Comment by howardjohn** [comment]
Don't cherrypick it until master has been merged, its confusing for release managers and more work for you if you need to change the PR

File: pilot/pkg/bootstrap/server.go [hunk 1]
```diff
@@ -608,7 +608,7 @@ func (s *Server) initSecureGrpcServer(port string, keepalive *istiokeepalive.Opt
 
 	cfg := &tls.Config{
 		Certificates: []tls.Certificate{certP},
-		ClientAuth:   tls.VerifyClientCertIfGiven,
+		ClientAuth:   tls.NoClientCert,
```
Comment (brian-avery) [inline pilot/pkg/bootstrap/server.go:611]: @costinm @myidpt @howardjohn Can you please address your thoughts on this? The original PR has unresolved issues concerning this. On top of that, the original PR was never merged, so this isn't really a "cherrypick".

**Comment by myidpt** [comment]
Abandon this one since we decided to not go in this path.

**Pull request closed without merging**